Samples/DirectMLNpuInference/main.cpp

```diff
@@ -34,32 +66,52 @@ void InitializeDirectML(ID3D12Device1** d3dDeviceOut, ID3D12CommandQueue** comma
     ComPtr<IDXCoreAdapter> adapter;
     if (factory)
     {
-        const GUID dxGUIDs[] = { DXCORE_ADAPTER_ATTRIBUTE_D3D12_CORE_COMPUTE };
-        ComPtr<IDXCoreAdapterList> adapterList;
-        THROW_IF_FAILED(factory->CreateAdapterList(ARRAYSIZE(dxGUIDs), dxGUIDs, IID_PPV_ARGS(&adapterList)));
-        for (uint32_t i = 0, adapterCount = adapterList->GetAdapterCount(); i < adapterCount; i++)
+        // If there's any required attributes, save time by only passing in required attributes instead.
+        std::vector<GUID> iteratingAdapterList = dxGuidRequireAllAttributes.empty() ? dxGuidAllowedAttributes : dxGuidRequireAllAttributes;
+        
+        for (auto& allowedGuid : iteratingAdapterList)
         {
-            ComPtr<IDXCoreAdapter> currentGpuAdapter;
-            THROW_IF_FAILED(adapterList->GetAdapter(static_cast<uint32_t>(i), IID_PPV_ARGS(&currentGpuAdapter)));
-
-            if (!forceComputeOnlyDevice && !forceGenericMLDevice)
-            {
-                // No device restrictions
-                adapter = std::move(currentGpuAdapter);
-                break;
-            }
-            else if (forceComputeOnlyDevice && currentGpuAdapter->IsAttributeSupported(DXCORE_ADAPTER_ATTRIBUTE_D3D12_CORE_COMPUTE))
-            {
-                adapter = std::move(currentGpuAdapter);
-                break;
-            }
-            else if (forceGenericMLDevice && currentGpuAdapter->IsAttributeSupported(DXCORE_ADAPTER_ATTRIBUTE_D3D12_GENERIC_ML))
+            if (adapter != nullptr) continue;
```

Comment:
>continue [](http://example.com/codeflow?start=36&length=8)

Should we keep looping if we have already found an adapter that meet our requirements?